framework/tlvf/tlvf.py

```diff
@@ -445,12 +445,22 @@ def generateObject(self, obj_meta, dict_value):
                 if param_meta.error: self.abort(param_meta.error)
                 param_type = param_meta.type
                 param_type_info = param_meta.type_info
+                is_enum_type = False
                 if param_type and param_type.startswith("e"):
```

Comment:
this line should be replaced with:
`if TypeInfo(param_type).type == TypeInfo.ENUM`
That provides you the functionality you implemented in the following lines.